### DISCUSSION TITLE
System.InvalidCastException: 'Unable to cast object of type 'System.Reflection.RuntimeMethodInfo' to type 'Mono.Cecil.IMethodSignature'.'

### DISCUSSION THREAD

**Discussion by HPXCoder** [jbevain/cecil#829] [Q&A] [answered]
iam making a program that edit IL of methods body in a dll 
and i got this error when saving the dll
how can i know the specific method that have this error and how to solve it 
![Pic](https://user-images.githubusercontent.com/31288493/152071838-ff5d799f-413e-4b60-8fee-9f1ecac58775.PNG)

**Comment by jbevain** [comment]
At this point your best bet is either to:

1) Audit your code doing the IL injection. Basically it looks like you're using the result of a `GetMethod` on a `System.Type` directly instead of calling `ImportReference` on the MethodInfo to get an object usable in Cecil's type model.
2) Compile Cecil alongside with your application, so when the exception is thrown you'll be able to see from the callstack what Cecil is doing at this point in time.

**Reply by HPXCoder** [reply]
oh really i appreciate your help i do what did you say and i got anthor error
could you help please?
![Pic2](https://user-images.githubusercontent.com/31288493/152084765-6dd8b9c8-e6bb-46b1-8b7f-bf8fc842f31e.PNG)

**Reply by HPXCoder** [reply]
how can i import any type to my module :D
i tried 
`mymodule.ImportReference(resolved.DeclaringType);`

resolved.DeclaringType is the member reference of System.string Concat

and tried 

`mymodule.Import(resolved.DeclaringType);`

no thing work

**Comment by jbevain** [comment, marked as answer]
This next error you're seeing in different.

It means you're using a MethodReference that was created in one module inside another.

You need to make sure all the references you use is one module has been properly created through `ImportReference` into the module you're modifying.

**Reply by HPXCoder** [reply]
Thank You :D
